code/game/objects/items/weapons/RCD.dm

```diff
@@ -10,6 +6,205 @@
 #define MODE_WINDOW		"Windows"
 #define MODE_DECON		"Deconstruction"
 
+/// A generic action for an RCD.
+/datum/rcd_act
+	/// How much compressed matter this action costs.
+	var/cost = 0
+	/// How long this action takes.
+	var/delay = 0
+	/// The message (if any) to send the user when the action starts.
+	var/start_message
+	/// The effect (if any) to create when the action starts.
+	var/start_effect_type
+	/// The effect (if any) to create when the action completes.
+	var/end_effect_type
+	/// The mode the RCD must be in.
+	var/mode
+
+/// Attempt the action. This should not need to be overridden.
+/datum/rcd_act/proc/try_act(atom/A, obj/item/rcd/rcd, mob/user)
+	if(!can_act(A, rcd, user))
+		return FALSE
+	// We don't use the sound effect from use_tool because RCDs have a different sound effect for the start and end.
+	playsound(rcd.loc, 'sound/machines/click.ogg', 50, TRUE)
```

Comment:
```suggestion
	playsound(get_turf(rcd), 'sound/machines/click.ogg', 50, TRUE)
```
We generally do local sounds at the turf so that weird location issues don't happen